homeassistant/components/kegtron/sensor.py

```diff
@@ -32,44 +32,53 @@
 from .device import device_key_to_bluetooth_entity_key
 
 SENSOR_DESCRIPTIONS = {
-    KegtronSensorDeviceClass.PORT_COUNT: SensorEntityDescription(
+    (KegtronSensorDeviceClass.PORT_COUNT, None): SensorEntityDescription(
```

Comment:
We should not make tuples here to extend information, instead extend the entity description itself.

Reply:
A port count does not have a unit (it's just an integer). I'm not sure how the entity description should be extended in this case, considering e.g. https://github.com/home-assistant/core/pull/81844 where explicit `native_unit_of_measurement=None`s were removed...

Making this change here makes removing some 600 lines of code possible in #82418.

Reply:
An entity description can be extended in the integration and add additional properties specific to this integration. This is a very normal and common pattern in Home Assistant, please use that instead of the current solution.

../Frenck

Reply:
In this end, this PR is duplicating code actually, as the property is already present.

Reply:
Considering this in the PR description:

> This is so they're in sync with the other sensor_state_data based sensor description dicts

I would say others should be adjusted too. If this is a common pattern, we need to get rid of the definition duplication.

Reply:
> An entity description can be extended in the integration and add additional properties specific to this integration. 

This isn't adding new (or removing existing) specific properties, this is just noting the unit or lack thereof in the dict's key. The tuple key is the norm in the rest of the passive BLE integrations, so adjusting this here would let the same (future, from #82418) function
https://github.com/home-assistant/core/blob/96ccb4c7c6aa2fbf6d760fcd787964e969e75dfd/homeassistant/helpers/bluetooth.py#L34-L39
that's now duplicated about 14 to 15 times be used for all passive BLE integrations.

> In this end, this PR is duplicating code actually, as the property is already present. We should not duplicate property data here.
> I would say others should be adjusted too. If this is a common pattern, we need to get rid of the definition duplication.

Yeah, in a more perfect world this mapping wouldn't even need to exist (or at the very least it shouldn't need keys), since the HA entity description should be derivable from the sensor-state-data data. That could maybe be done if #82418 is merged, so it doesn't need to be done for N copies of the same function.

Reply:
> This isn't adding new (or removing existing) specific properties

Right, which is why this PR  doesn't make sense. It adds definitions to a tuple, that are duplicate of the existing entity descriptions.

> The tuple key is the norm in the rest of the passive BLE integrations

Sounds off to me. We should not have duplicate definitions.

> that's now duplicated about 14 to 15 times be used for all passive BLE integrations.

Yeah, that is not wished for. Lets remove those other occurrences instead of adding more.

> Yeah, in a more perfect world this mapping wouldn't even need to exist 

Then that should be the goal and not this PR.

> That could maybe be done if https://github.com/home-assistant/core/pull/82418 is merged

I don't see how that is related at all. I think the de-depulication done in #82418 isn't wished for either. DRY is, in many cases, our enemy.